### PR TITLE
Don't error when the push is too big

### PR DESCRIPTION
**Pull request by pbiggar**
Large pushes are happening because the user is renaming something (probably a DB), that causes a ton of other TLs to be pushed.

For now, it's much much easier to drop these. I'd rather put the effort into making renames not change every toplevel, and this happens rarely and canvases are fairly rarely used by two people at once anyway.

### CODE REVIEW

File: fsharp-backend/src/LibBackend/Pusher.fs [hunk 1]
```diff
@@ -69,10 +72,19 @@ let pushNew404 (canvasID : CanvasID) (f404 : TraceInputs.F404) =
...
+type AddOpEventTooBigPayload = { tlids : List<tlid> }
 
 // For exposure as a DarkInternal function
 let pushAddOpEvent (canvasID : CanvasID) (event : Op.AddOpEvent) =
-  push canvasID "add_op" (Json.OCamlCompatible.serialize event)
+  let payload = Json.OCamlCompatible.serialize event
+  if String.length payload > 10240 then
+    let tlids = List.map LibExecution.OCamlTypes.tlidOf event.``params``.ops
+    let tooBigPayload = { tlids = tlids } |> Json.Vanilla.serialize
+    // push canvasID "addOpTooBig" tooBigPayload
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/Pusher.fs:83]: do you need to uncomment this?

Comment (pbiggar) [reply fsharp-backend/src/LibBackend/Pusher.fs:83]: No, that was the point of the comment that immediately follows.

I guess I should make that clearer?

Comment (StachuDotNet) [reply fsharp-backend/src/LibBackend/Pusher.fs:83]: My interpretation of the line is "push a notification that there was some add_op involving so many canvases that the serialized payload would be huge." That seems good+safe to have even with no receiver? (i.e. so we can use that data for some analysis, if pushes are persisted?)

Comment (pbiggar) [reply fsharp-backend/src/LibBackend/Pusher.fs:83]: I think it'll make a BIG RED ERROR BAR with no receiver, which the user can't do anything about.

I'll fix the comment.

Comment (StachuDotNet) [reply fsharp-backend/src/LibBackend/Pusher.fs:83]: Ah, ok. Maybe we should (separately) _not_ raise a big red error bar when there's no receiver?
That sounds like a reasonable way to tee up something to be handled later, recording along the way for potential analysis.

Comment (StachuDotNet) [reply fsharp-backend/src/LibBackend/Pusher.fs:83]: PR looks good with comment update.

**Review by StachuDotNet — APPROVED** [review]

File: fsharp-backend/src/LibBackend/Pusher.fs (hunk 1 above)
Comment (pbiggar) [reply fsharp-backend/src/LibBackend/Pusher.fs:86]: Thanks

**Pull request merged**